### PR TITLE
Enable test coverage reporting with Cobertura and CodeClimate

### PR DESCRIPTION
**Pull request by john-odonnell**
### Desired Outcome

Report Go unit test coverage of `pkg/` and `cmd/` files to Cobertura and CodeClimate.

### Implemented Changes

- Add Go test coverage tools to `Dockerfile.junit`
- Update `bin/test` to convert Go coverage report to format consumable by Cobertura
- Add CodeClimate reporting to `Jenkinsfile` 
- Use `go test` flag `-coverpkg`, which allows tests to record coverage for source files in other packages.
  - Some packages use a `tests` sub-package for test files, which leaves the parent package with 0% coverage.
  - Some packages, like `common` and `config`,  should register coverage from tests in packages that use their common functionality.

See: [Resulting Cobertura coverage report in Jenkins](https://jenkins.conjur.net/job/cyberark--conjur-authn-k8s-client/job/enable-coverage-report/9/cobertura/)

### Connected Issue/Story

CyberArk internal issue link: [ONYX-14776](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-14776)

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

File: Dockerfile.junit [hunk 1]
```diff
@@ -2,7 +2,7 @@ FROM golang:1.17-alpine
 MAINTAINER CyberArk Software Ltd.
 LABEL id="conjur-authn-k8s-client-junit-processor"
 
-WORKDIR /test
+WORKDIR /conjur-authn-k8s-client/test
```
Comment (doodlesbykumbi) [inline Dockerfile.junit:5]: Why is this change needed ?

**Review by doodlesbykumbi — APPROVED** [review]
LGTM. Left a couple curiosity questions

File: bin/test [hunk 2]
```diff
@@ -25,14 +28,15 @@ set -e
...
 
 echo "Creating junit report..."
 
 docker run --rm \
-  -v $PWD/test:/test \
+  -v "$PWD"/:/conjur-authn-k8s-client/ \
   conjur-authn-k8s-client-junit:latest \
   bash -exc "
     cat ./junit.output | go-junit-report > ./junit.xml
+    cat ./c.out.tmp | grep -v authenticator_test_server.go > ./c.out
```
[the remark this answers is not in the document]
Comment (john-odonnell) [reply bin/test:40]: `authenticator_test_server.go` is a mock server used in testing, but is counted toward our coverage targets as it is considered a source file. This line removes mention of the file from the coverage report, so it's ignored when calculating coverage.

File: Dockerfile.junit (hunk 1 above)
Comment (john-odonnell) [reply Dockerfile.junit:5]: This Dockerfile is used to convert the standard `go test` output to formats consumable by Cobertura and CodeClimate. That said, `gocov-xml` was failing when it didn't have access to the project's source files:

```
error: can't find "main.go": no required module provides package github.com/cyberark/conjur-authn-k8s-client/cmd/authenticator: go.mod file not found in current directory or any parent directory; see 'go help modules'
```
